ospp/OSPP-2025.md

```diff
@@ -4,3 +4,77 @@ Community Contact: @RuoqingHe
 
 ## Accepted Projects
 
+# Project Name
+
+GStreamer Backend for vhost-device-sound
+
+## Project Mentor
+
+Dorinda Bassey <dbassey@redhat.com>, Matias Ezequiel Vara Larsen <mvaralar@redhat.com>
+
+## Project Description
+
+virtio-sound device emulation has recently been developed in the Rust [vhost-device-sound](https://github.com/rust-vmm/vhost-device/tree/main/vhost-device-sound) crate.
+The crate currently contains audio backends for the ALSA and PipeWire sound APIs. The aim of this project is to build a new
+[GStreamer](https://gstreamer.freedesktop.org/documentation/?gi-language=c) audio backend.
+
+Audio backends are written by implementing the [AudioBackend trait](https://github.com/rust-vmm/vhost-device/blob/b113dc6b288ff2b75968e245a716dc7e1436aec1/vhost-device-sound/src/audio_backends.rs#L20).
+Refer to [alsa.rs](https://github.com/rust-vmm/vhost-device/blob/main/vhost-device-sound/src/audio_backends/alsa.rs) and
+[pipewire.rs](https://github.com/rust-vmm/vhost-device/blob/main/vhost-device-sound/src/audio_backends/pipewire.rs) for examples of existing backends.
+The [Stream](https://github.com/rust-vmm/vhost-device/blob/6ca911eb5f6a73c534e234eedea90cf26d1d71ad/vhost-device-sound/src/stream.rs#L174) and
+[Buffer](https://github.com/rust-vmm/vhost-device/blob/6ca911eb5f6a73c534e234eedea90cf26d1d71ad/vhost-device-sound/src/stream.rs#L242) structs are used to
+transfer audio samples between the virtio-sound device and the sound API (e.g. GStreamer).
+
+The backend should be implemented using the [GStreamer Rust bindings](https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/tree/main). Mono and stereo
+playback and capture should be supported. The GStreamer pipelines for playback and capture
+will be hardcoded and only Linux needs to be supported.
+
+## Project Notes
+
+- The project focuses on adding a GStreamer-based audio backend to the vhost-device-sound crate, which currently supports ALSA and PipeWire.
+- The implementation will follow the AudioBackend trait used by existing backends.
+- The GStreamer Rust bindings will be used to integrate the new gstreamer audio backend.
+- The backend will support mono and stereo playback and capture with hardcoded GStreamer pipelines.
+- The implementation will be tested using QEMU as the vhost-user frontend.
+- Automated tests will be written using cargo test.
+- A possible stretch goal is contributing to rust-vmm/vhost-device by fixing existing issues.
+
+## Project Difficulty
+
+Advanced (some understanding of QEMU pci device would be good to know)
```

Comment:
Perhaps more than QEMU (which in this project I don't think should be touched, but just used to test the work), I would mention that knowledge of audio devices or gstreamer is a plus, but absolutely not required. (To avoid to scare off potential contributors).

Reply:
Yeah, I think it might scare them off.